crates/router/src/routes/dummy_connector/utils.rs

```
@@ -309,12 +313,21 @@ impl types::DummyConnectorCard {
                     }),
                 ))
             }
-            "4000000000009995" => Ok(types::DummyConnectorCardFlow::NoThreeDS(
-                types::DummyConnectorStatus::Failed,
-                Some(errors::DummyConnectorErrors::PaymentDeclined {
-                    message: "Insufficient funds",
-                }),
-            )),
+            "4000000000009995" => {
+                if connector == DummyConnectors::StripeTest {
+                    Ok(types::DummyConnectorCardFlow::NoThreeDS(
+                        types::DummyConnectorStatus::Succeeded,
+                        None,
+                    ))
+                } else {
+                    Ok(types::DummyConnectorCardFlow::NoThreeDS(
+                        types::DummyConnectorStatus::Failed,
+                        Some(errors::DummyConnectorErrors::PaymentDeclined {
+                            message: "Internal Server Error",
```

Can we throw a better error message here?